common/parseRoute.js

```diff
@@ -0,0 +1,16 @@
+module.exports = path => {
+    let method = 'GET';
+    const methods = ['POST', 'GET', 'DELETE', 'PUT'];
+
+    methods.forEach(m => {
```

Comment:
this seems a little hacky but nothing explicitly wrong about. Thinking out loud: instead of changing `path`, return `path` and `originalPath` or something similar.